start_local_runner.sh

```diff
@@ -62,15 +63,15 @@ if [[ $# < 2 ]] ; then
 fi
 
 gitlab_url=$1
-runner_token=$2
+token=$2
 shift 2
 
 exec docker run --rm \
-	--entrypoint "/register_and_run.sh" \
 	-v /var/run/docker.sock:/var/run/docker.sock \
 	-e GITLABURL="$gitlab_url" \
-	-e RUNNERTOKEN="$runner_token" \
+	-e TOKEN="$token" \
```

Comment:
I wonder if there is a way to avoid putting the token in plain text on the command line? Perhaps something with secrets?   Though that may then be very swarm-specific, and more trouble than it's worth.
(Also we've always had the token on the command line before, so no need to hold up merging on account of this question)

Reply:
It would be pretty trivial to read the TOKEN from a secret file mounted in to the container, however I don't think this would noticeably increase security. Regardless of how securely we ended up getting the TOKEN value in to a container, each container is mounted with write access to the hosts docker socket. Every runner would be able to discover the token of any other runner by doing a `docker inspect <other-runner>` and following the information there. The runners will start other containers as part of CI, but these other containers won't inherit the environment variables or the secrets. The other containers may have a docker socket mounted presenting the same `docker inspect` hole as the runner.

Reply:
And specifically with this `start_local_runner.sh` script, it is explicitly for debugging pipeline issues and testing changes to the runner image, not for production use. Having a token in the command line on your local development machine is less of an issue - if you can't trust your development environment then you've got bigger issues!

Reply:
All good - I agree with you on the relative security